SparkConnector/sparkconnector/configuration.py

```diff
@@ -132,11 +132,15 @@ def configure(self, opts, ports):
 
         # Extend conf adding logging of log4j to java options
         base_extra_java_options = "-Dlog4j.configuration=file:%s" % self.connector.log4j_file
-        extra_java_options = conf.get("spark.driver.extraJavaOptions")
-        if extra_java_options:
-            extra_java_options = base_extra_java_options + " " + extra_java_options
-        else:
-            extra_java_options = base_extra_java_options
+        extra_java_options = conf.get("spark.driver.extraJavaOptions", "")
+
+        # SPARK_DRIVER_EXTRA_JAVA_OPTIONS contains the default value of spark.driver.extraJavaOptions.
```

Comment:
I propose to replace this comment and use the one below for the whole block, starting with the first setting of extra_java_options:
```
# Fill extra_java_options as a concatenation of:
1. Options configured by the user via the UI of SparkConnector
2. SPARK_DRIVER_EXTRA_JAVA_OPTIONS variable, set for example to load required NXCALS settings
3. Base Java options with log4j configuration
```

Reply:
1 user_extra_java_options
2 env_extra_java_options
3 logging_extra_java_options